src/Makefile.am

```diff
@@ -51,18 +51,8 @@ lib_LTLIBRARIES += \
 	libnss_mdns6_minimal.la
 endif
 
-noinst_PROGRAMS = nss-test
-sources = util.c util.h
-
-if ENABLE_AVAHI
-sources += avahi.c avahi.h
-noinst_PROGRAMS += avahi-test
-endif
-
-if ENABLE_LEGACY
-sources += query.c query.h dns.c dns.h
-noinst_PROGRAMS += mdns-test
-endif
+noinst_PROGRAMS = nss-test avahi-test
```

Comment:
you want check_PROGRAMS.  noinst get built when you run `make`, but check get built only when you `make check`.